### DISCUSSION TITLE
Static export translations different redirects depending on domain

### DISCUSSION THREAD

**Discussion by joaopedrodcf** [amannn/next-intl#499] [Q&A] [answered]
First thanks for creating such and awesome library 💯 

So I was trying out the library for a static exports project and from the docs it's suggested when the user enters the / we redirect him to the /en and this makes perfect sense 💯 

https://next-intl-docs.vercel.app/docs/routing/middleware#usage-without-middleware-static-export

But for my use case I will deploy my application for different localizations and I wanted to redirect for different locales depending on the domain we enter, example if I go to www.my-site.fr I want to go to www.my-site.fr/fr.

With this the depending on the domain the user should land in a different locale, and be able to change it as normal.

Maybe the only solution is really going to SSR 🤔 

Any ideas ? thanks in advance 💯

**Comment by amannn** [comment, marked as answer]
Thank you for the kind words! 😊 

I think for now your only option is to go with a server, that's correct!

Is the motivation for the static export a cost question? I'm just curious 🙂

**Reply by joaopedrodcf** [reply]
Yeah it was mostly because of costs 😢 

One thing I thought could work was to instead of having the root layout as a server component make it a client component and redirect in the client, it worked well locally but when serving the assets it just breaks 😢 

When I serve the assets with npx serve the site only works on the / but if I try other routes it just doesn't work 😢 

```ts
import { usePathname, useRouter } from 'next/navigation';
...

export default function RootLayout({ children }: Props) {
    const router = useRouter();
    const pathname = usePathname();

    useEffect(() => {
        const domain = window.location.hostname;
        const config = ...

        if (pathname === '/') {
            router.replace(config,defaultLocale);
        }
    }, []);

    return (
        <html lang="en">
            <body>{children}</body>
        </html>
    );
}
```

Thanks for the answer 🥇
